packages/caravan-wallets/src/bcur2/decoder.ts

```diff
@@ -37,18 +37,25 @@ export class BCURDecoder2 {
 
   private progress: string = "Idle";
 
+  private isInjectedDecoder: boolean;
+
   /**
    * Creates a new BCUR2 decoder instance
+   * @param decoder - Optional URRegistryDecoder instance. If not provided, creates a new one.
    */
-  constructor() {
-    this.decoder = new URRegistryDecoder();
+  constructor(decoder?: URRegistryDecoder) {
+    this.decoder = decoder || new URRegistryDecoder();
+    this.isInjectedDecoder = Boolean(decoder);
   }
 
   /**
    * Resets the decoder state to initial values
    */
   reset() {
```

Comment:
Where is reset used? It feels a bit weird to have this extra property to track. Maybe reset can also support dependency injection and it can be passed in as an argument with the argument defaulting to a new one passed in?